backend/src/ml_space_lambda/utils/common_functions.py

```diff
@@ -200,16 +200,23 @@ def generate_html_response(status_code, response_body):
 
 def generate_exception_response(e):
     status_code = 400
+    error_msg = str(e)
     if hasattr(e, "response"):  # i.e. validate the exception was from an API call
         metadata = e.response.get("ResponseMetadata")
         if metadata:
             status_code = metadata.get("HTTPStatusCode", 400)
+        if e.response["Error"]["Code"] in ["ResourceInUse", "ValidationException"] and any(
+            error in error_msg for error in ["Cannot create a duplicate", "already existing", "already exists"]
+        ):
+            e = f"The resource name you provided already exists. Please choose a different name. Full message: {error_msg}"
+        elif e.response["Error"]["Code"] in ["ResourceLimitExceeded"]:
+            e = f"You have reached the maximum allowed usage for this resource. Please contact your MLSpace administrator to increase the allowed usage limits. Full message: {error_msg}"
+
```

Comment:
This is great! It might make sense to break out these errors though so it is more cleanly extensible in the future. Then we could just loop over our known errors a little more cleanly. Then our error checking logic never gets more complicated, we just add to our `ERROR_MAP` structure.

```python
ERROR_MAP = [{
  Type: "duplicate"
  Codes: ["ResourceInUse", "ValidationException"],
  MatchStrings: ["Cannot create a duplicate", "already existing", "already exists"],
  FriendlyMessage: "The resource name you provided already exists. Please choose a different name."
 }, {
  Type: "limit"
  Codes: ["ResourceLimitExceeded"],
  # hack to always match, could add logic to check if MatchStrings is empty or not
  MatchStrings: [""],
  FriendlyMessage: "You have reached the maximum allowed usage for this resource. Please contact your MLSpace administrator to increase the allowed usage limits."
}]
...

for errorType in ERROR_MAP:
  if e.response["Error"]["Code"] in errorType["Codes"] and any(error in error_msg for error in errorType["MatchStrings"]):
    e = f"{errorType["FriendlyMessage"]} Full message: {error_msg}"
```

Reply:
I like it! Ill add this